stumpy/mmotifs.py

```diff
@@ -0,0 +1,187 @@
+
+    Parameters
+    ----------
+    T: numpy.ndarray
+        The multi-dimensional time series or sequence
+
+    P: numpy.ndarray
+        Multi-dimensional Matrix Profile of T
+
+    I: numpy.ndarray
+        Multi-dimensional Matrix Profile indices
+
+    min_neighbors : int, default 1
+        The minimum number of similar matches a subsequence needs to have in order
+        to be considered a motif. This defaults to `1`, which means that a subsequence
+        must have at least one similar match in order to be considered a motif.
+
+    max_distance: flaot, default None
+        Maximal distance that is allowed between a query subsequence
+        (a candidate motif) and all subsequences in T to be considered as a match.
+        If None, this defaults to
+        `np.nanmax([np.nanmean(D) - 2 * np.nanstd(D), np.nanmin(D)])`
+        (i.e. at least the closest match will be returned).
+
+    cutoff: numpy.ndarray or float, default None
+        The largest matrix profile value (distance) for each dimension of the
+        multidimensional matrix profile that a multidimenisonal candidate motif is
+        allowed to have.
+        If cutoff is only one value, these value will be applied to every dimension.
+
+    max_matches: int, default 10
+        The maximum amount of similar matches (nearest neighbors) of a motif
+        representative to be returned.
+        The first match is always the self-match for each motif.
+
+    max_motifs: int, default 1
+        The maximum number of motifs to return
+
+    atol : float, default 1e-8
+        The absolute tolerance parameter. This value will be added to `max_distance`
+        when comparing distances between subsequences.
+
+    Returns
+    -------
+    motif_matches_distances: numpy.ndarray
+        The distances corresponding to a set of subsequence matches for each motif.
+
+    motif_matches_indices: numpy.ndarray
+        The indices corresponding to a set of subsequences matches for each motif.
+
+    motif_subspaces: numpy.ndarray
+        A numpy.ndarray consisting of arrays that contain the `k`-dimensional
+        subspace for each motif.
+
+    motif_mdls: numpy.ndarray
+        A numpy.ndarray consisting of arrays that contain the mdl results for
+        finding the dimension of each motif
+    """
+    if max_motifs < 1:  # pragma: no cover
+        logger.warning(
+            "The maximum number of motifs, `max_motifs`, "
+            "must be greater than or equal to 1"
+        )
+        logger.warning("`max_motifs` has been set to `1`")
+        max_motifs = 1
+
+    # Calculate subsequence length
+    T = core._preprocess(T)  # convert dataframe to array if necessary
+    m = T.shape[-1] - P.shape[-1] + 1
+
+    # Calculate exclusion zone
+    excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+
+    # Precompute rolling means and standard deviations
+    T, mean_T, sigma_T = core.preprocess(T, m)
+

```

Comment:
Is there a reason why you removed the warning:

```
if max_motifs < 1:  # pragma: no cover
        logger.warn(
            "The maximum number of motifs, `max_motifs`, "
            "must be greater than or equal to 1"
        )
        logger.warn("`max_motifs` has been set to `1`")
        max_motifs = 1
```

Is it no longer needed?

Reply:
It still exists in my version of the code in line 79 - 85.

Strange, it should be visible for you as well, or do our codes not match?

Maybe I did something wrong? Can you check?